[docs] Add initial documentation for read-write mode (#3951)
grafana/mimir: docs/sources/mimir/operators-guide/architecture/about-grafana-mimir-architecture/index.md

Grafana Mimir microservices are called components.

Grafana Mimir's design compiles the code for all components into a single binary.
The `-target` parameter controls which component(s) that single binary will behave as.

> For a simple way to get started, you can run Grafana Mimir in [monolithic mode]({{< relref "../deployment-modes/index.md#monolithic-mode" >}}), with all components running simultaneously in one process, or in [read-write mode]({{< relref "../deployment-modes/index.md#read-write-mode" >}}), which groups components into read, write, and backend paths.

Reviewer: ```suggestion
To get started easily, run Grafana Mimir in [monolithic mode]({{< relref "../deployment-modes/index.md#monolithic-mode" >}}) with all components running simultaneously in one process, or in [read-write mode]({{< relref "../deployment-modes/index.md#read-write-mode" >}}), which groups components into _read_, _write_, and _backend_ paths.
```

> For those looking for a simple way to get started, Grafana Mimir can also be run in [monolithic mode]({{< relref "../deployment-modes/index.md#monolithic-mode" >}}), with all components running simultaneously in one process, or in [read-write mode]({{< relref "../deployment-modes/index.md#read-write-mode" >}}), which groups components into their read and write paths.

Reviewer 2: WDYT about this?

```suggestion
For those looking for a simple way to get started, Grafana Mimir can also be run in [monolithic mode]({{< relref "../deployment-modes/index.md#monolithic-mode" >}}), with all components running simultaneously in one process, or in [read-write mode]({{< relref "../deployment-modes/index.md#read-write-mode" >}}), which groups components into read, write and backend paths.
```

> For those looking for a simple way to get started, Grafana Mimir can also be run in [monolithic mode]({{< relref "../deployment-modes/index.md#monolithic-mode" >}}), with all components running simultaneously in one process, or in [read-write mode]({{< relref "../deployment-modes/index.md#read-write-mode" >}}), which groups components into read, write and backend paths.

Reviewer: ```suggestion
For a simple way to get started, you can run Grafana Mimir in [monolithic mode]({{< relref "../deployment-modes/index.md#monolithic-mode" >}}), with all components running simultaneously in one process, or in [read-write mode]({{< relref "../deployment-modes/index.md#read-write-mode" >}}), which groups components into read, write, and backend paths.
```
